inferno/tests/test_net.py

```diff
@@ -424,6 +424,51 @@ def test_get_params_with_uninit_callbacks(self, net_cls, module_cls):
         net.initialize()
         net.get_params()
 
+    def test_call_fit_twice_retrains(self, net_cls, module_cls, data):
+        X, y = data[0][:100], data[1][:100]
+        net = net_cls(module_cls, cold_start=True).fit(X, y)
+        params_before = net.module_.parameters()
+
+        net.max_epochs = 0
+        net.fit(X, y)
+        params_after = net.module_.parameters()
+
+        assert len(net.history) == 0
+        for p0, p1 in zip(params_before, params_after):
+            assert (p0 != p1).data.any()
```

Comment:
I'm not sure what's tested here. We don't iterate in `fit_loop` but we call fit. Do we expect it to not blow up? Maybe add a comment here.